packages/pwa/app/components/_app-config/index.jsx

```diff
@@ -72,7 +53,13 @@ const AppConfig = ({children, locals = {}}) => {
 }
 
 AppConfig.restore = (locals = {}) => {
-    const locale = getLocale(locals) || DEFAULT_LOCALE
+    // TODO: There is probably a better way to do this. Maybe build it into the SDK too.
+    if (typeof window !== 'undefined') {
+        locals.originalUrl = window.location.pathname
+    }
+
+    const [locale] = AppConfig.getUserPreferredLocales(locals)
```

Comment:
@bendvc this locale, which will be passed in to our Commerce API client, cannot be equal to user preferred locales:

- First of all, `getUserPreferredLocales` is meant to return an array that can contain more than 1 locale (e.g. browser can send multiple preferred user locales). So in this line, we're extracting only the first item in the array.
- What the Commerce API client needs to know is the _resolved_ or target locale